### DISCUSSION TITLE
Master service vs node service

### DISCUSSION THREAD

**Discussion by lukaslau** [pschlan/cron-job.org#120] [Q&A]
1. What is the difference between master service (runs on 9091 port) and node service (runs on 9090 port)?
2. Is it needed manually to add an entry to master database's node table or it is added automatically?

@pschlan

**Comment by pschlan** [comment]
Hi!

1. The master service runs on a single machine only and provides access to configuration and user settings. It's also used to store stats from the nodes. For fault tolerance, it could be run on multiple systems accessing the same MySQL server (which might have its own redundance) with a failover IP. The node service (potentially) runs on multiple machines and is responsible for storing and executing actual jobs. You can scale the service by adding more machines running the node service.
2. Currently this has to be done manually, there's no auto-discovery.

Hope this helps!

Patrick
